### PR TITLE
fix: [CDS-73138]: add null check - which was breaking the UI, also added default value for allValues in case of template

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
#### Screenshots

Before: 

https://github.com/harness/harness-core-ui/assets/98325173/d22cca90-efe6-4758-8f22-79266c3d9ac0

After:

https://github.com/harness/harness-core-ui/assets/98325173/62b5264b-a604-469b-be96-de908506159b

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by litvinovstas — DISMISSED** [review]

**Review by sahithikolichala — DISMISSED** [review]

**Comment by sahithikolichala** [comment]
retrigger cypress-cv

**Comment by sahithikolichala** [comment]
Screen Shot 2023-06-27 at 1 14 53 PM
fixed to this

File: src/modules/70-pipeline/components/PipelineInputSetForm/ExecutionWrapperInputSetForm.tsx [hunk 1]
```diff
@@ -34,13 +34,13 @@ import stepCss from '@pipeline/components/PipelineSteps/Steps/Steps.module.scss'
...
+    if (item?.step?.identifier === stepId) {
       responseStep = item
-    } else if (item.stepGroup?.identifier === stepId) {
+    } else if (item?.stepGroup?.identifier === stepId) {
       responseStep = item
-    } else if (item.parallel) {
-      return item.parallel.forEach(node => {
-        if (node.step?.identifier === stepId || node.stepGroup?.identifier === stepId) {
+    } else if (item?.parallel) {
+      return item?.parallel.forEach(node => {
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineInputSetForm/ExecutionWrapperInputSetForm.tsx:42]: nit: Here we will always have item.parallel as optional chaining is handled in else if

**Review by PratyushGar — DISMISSED** [review]

**Review by tan-nhu — APPROVED** [review]

**Review by cjlee01 — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineInputSetForm/ExecutionWrapperInputSetForm.tsx (hunk 1 above)
Comment (Sagar-Rajput7) [reply src/modules/70-pipeline/components/PipelineInputSetForm/ExecutionWrapperInputSetForm.tsx]: done

**Comment by Sagar-Rajput7** [comment]
closing and reopening the PR to retrigger the sonar-cube check.

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by Sagar-Rajput7** [comment]
> **SonarQube Quality Gate**
> 
> [Quality Gate passed](https://sonar.harness.io/dashboard?id=nextgenui&pullRequest=16616)
> 
> [Bug](https://sonar.harness.io/project/issues?id=nextgenui&pullRequest=16616&resolved=false&types=BUG) [A](https://sonar.harness.io/project/issues?id=nextgenui&pullRequest=16616&resolved=false&types=BUG) [0 Bugs](https://sonar.harness.io/project/issues?id=nextgenui&pullRequest=16616&resolved=false&types=BUG) [Vulnerability](https://sonar.harness.io/project/issues?id=nextgenui&pullRequest=16616&resolved=false&types=VULNERABILITY) [A](https://sonar.harness.io/project/issues?id=nextgenui&pullRequest=16616&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonar.harness.io/project/issues?id=nextgenui&pullRequest=16616&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonar.harness.io/security_hotspots?id=nextgenui&pullRequest=16616) [A](https://sonar.harness.io/security_hotspots?id=nextgenui&pullRequest=16616) [0 Security Hotspots](https://sonar.harness.io/security_hotspots?id=nextgenui&pullRequest=16616) [Code Smell](https://sonar.harness.io/project/issues?id=nextgenui&pullRequest=16616&resolved=false&types=CODE_SMELL) [A](https://sonar.harness.io/project/issues?id=nextgenui&pullRequest=16616&resolved=false&types=CODE_SMELL) [0 Code Smells](https://sonar.harness.io/project/issues?id=nextgenui&pullRequest=16616&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonar.harness.io/component_measures?id=nextgenui&pullRequest=16616) No Coverage information [No Duplication information](https://sonar.harness.io/component_measures?id=nextgenui&pullRequest=16616&metric=duplicated_lines_density&view=list) No Duplication information

This sonar-cube report shows everything passing but not sure why the check is not detecting this.

**Pull request merged**